gtest_parallel.py

```diff
@@ -467,119 +467,130 @@ def start_daemon(func):
+                         'readable file. The format of the file is specified at '
+                         'https://www.chromium.org/developers/the-json-test-results-format')
+  parser.add_option('--timeout', type='int', default=None,
+                    help='Interrupt all remaining processes after the given '
+                         'time (in seconds).')
+
+  (options, binaries) = parser.parse_args()
+
+  if binaries == []:
+    parser.print_usage()
+    sys.exit(1)
+
+  if options.shard_count < 1:
+    parser.error("Invalid number of shards: %d. Must be at least 1." %
+                 options.shard_count)
+  if not (0 <= options.shard_index < options.shard_count):
+    parser.error("Invalid shard index: %d. Must be between 0 and %d "
+                 "(less than the number of shards)." %
+                 (options.shard_index, options.shard_count - 1))
+
+  # Check that all test binaries have an unique basename. That way we can ensure
+  # the logs are saved to unique files even when two different binaries have
+  # common tests.
+  unique_binaries = set(os.path.basename(binary) for binary in binaries)
+  assert len(unique_binaries) == len(binaries), (
+      "All test binaries must have an unique basename.")
+
+  # Remove files from old test runs.
+  if os.path.isdir(options.output_dir):
+    shutil.rmtree(options.output_dir)
+  # Create directory for test log output.
+  try:
+    os.makedirs(options.output_dir)
+  except OSError as e:
+    # Ignore errors if this directory already exists.
+    if e.errno != errno.EEXIST or not os.path.isdir(options.output_dir):
+      raise e
+
+  timeout = (DummyTimer() if options.timeout is None
+             else threading.Timer(options.timeout, sigint_handler.interrupt))
+
+  test_results = None
+  if options.dump_json_test_results is not None:
+    test_results = CollectTestResults(options.dump_json_test_results)
+
+  save_file = os.path.join(os.path.expanduser("~"), ".gtest-parallel-times")
+  times = TestTimes(save_file)
+  logger = FilterFormat(options.output_dir)
+
+  task_manager = TaskManager(times, logger, test_results)
+
+  tasks = find_tests(binaries, additional_args)
+  logger.log_tasks(len(tasks))
+  execute_tasks(tasks)
+
+  if task_manager.passed:
+    logger.move_to('passed', task_manager.passed)
+    if options.print_test_times:
+      logger.print_tests('PASSED TESTS', task_manager.passed)
+
+  if task_manager.failed:
+    logger.print_tests('FAILED TESTS', task_manager.failed)
+    logger.move_to('failed', task_manager.failed)
+
+  if task_manager.started:
+    logger.print_tests('INTERRUPTED TESTS', task_manager.started.values())
+    logger.move_to('interrupted', task_manager.started.values())
+
+  logger.flush()
+  times.write_to_file(save_file)
+  if test_results:
+    test_results.dump_to_file_and_close()
+
+  if sigint_handler.got_sigint():
+    task_manager.global_exit_code = -signal.SIGINT
+  sys.exit(task_manager.global_exit_code)
+
+if __name__ == '__main__':
```

Comment:
Suggesting that you remove this __main__ check in here and only use gtest-parallel as the launcher. Up to you though.